draft-ietf-tls-ticketrequests.md

```diff
@@ -124,32 +152,91 @@ Clients MAY send this extension in ClientHello. It contains the following struct
 
 ~~~
 struct {
-    uint8 count;
+    uint8 new_session_count;
+    uint8 resumption_count;
 } TicketRequestContents;
 ~~~
 
-count
-: The number of tickets desired by the client.
-
-Clients can use TicketRequestContents.count to indicate the number of tickets
-they would prefer to receive. Servers SHOULD NOT send more tickets than
-TicketRequestContents.count, as clients will most likely discard any additional
-tickets. Servers SHOULD additionally place a limit on the number of tickets
-they are willing to send, to save resources. Therefore, the number of
-NewSessionTicket messages sent will typically be the minimum of the server's self-imposed
-limit and TicketRequestContents.count.
-
-A server that supports ticket requests MAY echo the "ticket_request" extension in the
-EncryptedExtensions message. If present, it contains a TicketRequestContents structure, where
-TicketRequestContents.count indicates the number of tickets the server expects to send to the client.
+new_session_count
+: The number of tickets desired by the client when the server chooses to
+negotiate a fresh session (full handshake).  This is also the ticket count
+requested when the client presented no initial ticket making a full handshake
+unavoidable.
+
+resumption_count
+: The number of tickets desired by the client when the server is willing to
+resume using the presented ticket.
+
+Clients can use the above structure to indicate the number of tickets they
+would prefer to receive when sessions are resumed or new sessions are
+negotiated.
+
+Typically, once a client's ticket cache is primed, a resumption count of 1 is a
+good choice that allows the server to replace each ticket with a fresh ticket,
+without over-provisioning the client with excess tickets.  However, clients that
+are racing multiple connections placing a separate ticket in each, will
+ultimately end up with just the tickets from a single resumed session, so
+in that case a resumption_count commensurate with the number of parallel
+sessions would be used.
+
+Priming of the ticket cache need not happen all at once.  Clients that
+anticipate opening frequent connections to a given server can ask for 2 or 3
+tickets on each full handshake, and attain the requisite steady-state cache
+size after a few initial full handshakes, without substantial risk of
+"overshoot" or need to accurately predict the connection concurrency.
+
+When a client presenting a previously obtained ticket finds that the server
+nevertheless negotiates a fresh session, the client should assume that any
+other tickets associated with the same session as the presented ticket are also
+no longer valid for resumption.  This includes not only the tickets obtained
+during the initial full handshake, but all tickets subsequently obtained as
+part of subsequent resumptions.  Requesting more than one ticket in case a
+full handshake is forced by the server helps to keep the session cache primed.
```

Comment:
```suggestion
no longer valid for resumption.  This includes tickets obtained
during the initial full handshake and all tickets subsequently obtained as
part of subsequent resumptions.  Requesting more than one ticket in case a
full handshake is forced by the server helps to keep the session cache primed.
```